app/uk/gov/hmrc/transitmovementsrouter/services/RoutingService.scala

```diff
@@ -87,39 +88,33 @@ class RoutingServiceImpl @Inject() (messageConnectorProvider: EISConnectorProvid
     movementType: MovementType,
     movementId: MovementId,
     messageId: MessageId,
-    messageType: MessageType,
+    messageType: RequestMessageType,
     payload: Source[ByteString, _]
   )(implicit hc: HeaderCarrier): EitherT[Future, RoutingError, Unit] = {
 
-    val officeOfDeparture       = payload.toMat(officeOfDepartureSink(messageType))(Keep.right).run()
+    val materializedResult      = payload.runWith(officeSink(messageType))
     implicit val messageSender  = MessageSender(movementId, messageId)
     implicit val updatedPayload = payload.via(buildMessage(messageType, messageSender))
 
-    EitherT(post(officeOfDeparture))
-  }
-
-  private def post(
-    office: Future[ParseResult[OfficeOfDeparture]]
-  )(implicit messageSender: MessageSender, body: Source[ByteString, _], hc: HeaderCarrier): Future[Either[RoutingError, Unit]] = {
-    import concurrent.ExecutionContext.Implicits.global
-    office.flatMap {
-      case Right(office) =>
-        postToEis(office).map {
-          case Right(_)    => Right(())
-          case Left(error) => Left(error)
-        }
-      case Left(routingError) =>
-        logger.error(s"Unable to extract office of departure: $routingError")
-        Future.successful(Left(routingError));
+    val maybeACustomsOffice = {
+      import concurrent.ExecutionContext.Implicits.global
```

Comment:
Rather than importing this here, we should implicitly pass the `ExecutionContext` down.

Reply:
added as an implicit.